drivers/net/ethernet/intel/idpf/idpf_eth_idc.h

```diff
@@ -0,0 +1,199 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/* Copyright (C) 2024 Intel Corporation */
+
+#ifndef _IDPF_ETH_IDC_H_
+#define _IDPF_ETH_IDC_H_
+
+#include <linux/auxiliary_bus.h>
+#include "idpf_eth_common.h"
+
+struct idpf_adapter;
+struct idpf_vc_xn_params;
+
+#define IDPF_MAX_Q				16
+#define IDPF_MIN_Q				2
+
+/**
+ * enum idpf_vport_type
+ * @IDPF_DEFAULT_VPORT: Default vport
+ * @IDPF_DYNAMIC_VPORT: Dynamic vport
+ */
+enum idpf_vport_type {
+	IDPF_DEFAULT_VPORT = 1,
+	IDPF_DYNAMIC_VPORT
+};
+
+/**
+ * enum idpf_queue_type
+ * @IDPF_QUEUE_DEFAULT_VPORT: Queue for Default vport
+ * @IDPF_QUEUE_DYNAMIC_VPORT: Queue for Dynamic vport
+ */
+enum idpf_queue_type {
+	IDPF_QUEUE_DEFAULT_VPORT = 1,
+	IDPF_QUEUE_DYNAMIC_VPORT
+};
+
+/**
+ * enum idpf_eth_idc_event_type
+ * Event type IDPF_ETH_IDC_EVENT_ALL_VPORTS are routed to all ports, while
+ * the event type IDPF_ETH_IDC_EVENT_SINGLE_VPORT is routed to only the
+ * intended port.
+ */
+enum idpf_eth_idc_event_type {
+	IDPF_ETH_IDC_EVENT_ALL_VPORTS,
+	IDPF_ETH_IDC_EVENT_SINGLE_VPORT
+};
+
+enum idpf_eth_idc_event_code {
+	/* Following events are from main driver to aux/eth driver */
+	IDPF_ETH_IDC_EVENT_LINK_CHANGE,
+	IDPF_ETH_IDC_EVENT_RESET_INITIATED,
+	IDPF_ETH_IDC_EVENT_RESET_COMPLETE,
+	IDPF_ETH_IDC_EVENT_DEINIT,
+	IDPF_ETH_IDC_EVENT_REMOVE,
+	IDPF_ETH_IDC_EVENT_PROBE_POST_INIT,
+
+	/* Following requests are from auxiliary eth driver to main driver */
+	IDPF_ETH_IDC_EVENT_REQ_HARD_RESET,
+	IDPF_ETH_IDC_EVENT_ENTER_PROMISCUOUS_MODE,
```

Comment:
PCI driver should not deal with any network specific functionality. So not sure why this PROMISCUOUS event is required??

Reply:
addressed in current changes.